File: tests/fixtures/semgrep-offline/negative/safe.rb
require 'digest'
require 'json'
require 'yaml'

def command_with_argv(user_input)
  system("ls", user_input)
end

def parse_data(payload)
  JSON.parse(payload)
end

def safe_deserialization(blob)
  YAML.safe_load(blob)
end

def strong_hash(data)
  Digest::SHA256.hexdigest(data)
end

File: tests/fixtures/semgrep-offline/negative/safe.js
/* The safe counterpart of every positive JavaScript case. Nothing may fire. */
const child_process = require('child_process');
const crypto = require('crypto');

function commandWithArgv(args) { child_process.execFile('/usr/bin/git', args); }
function parseData(input) { return JSON.parse(input); }
function namedHandler() { return function handler() { return 1; }; }
function strongHash(data) { return crypto.createHash('sha256').update(data); }
function unpredictable() { return crypto.randomBytes(32); }
function textSink(el, value) { el.textContent = value; }
